server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -492,17 +493,23 @@ static OverLimitStrategy createOverLimitStrategy(boolean trackRealMemoryUsage) {
         if (trackRealMemoryUsage && jvmInfo.useG1GC().equals("true")
         // messing with GC is "dangerous" so we apply an escape hatch. Not intended to be used.
             && Booleans.parseBoolean(System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.enabled"), true)) {
-            TimeValue lockTimeout = TimeValue.timeValueMillis(
-                Integer.parseInt(System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.lock_timeout_ms", "500"))
+
+            long lockTimeoutInMillis = Integer.parseInt(
+                System.getProperty("es.real_memory_circuit_breaker.g1_over_limit_strategy.lock_timeout_ms", "500")
             );
+            TimeValue lockTimeout = TimeValue.timeValueMillis(lockTimeoutInMillis);
+            // TODO: consider a separate property for the lock timeout under full GC and/or a ratio
```

Comment:
can this TODO be removed.

Reply:
Yes, I made a separate note in case we want to review this later